tools/README.md

```diff
@@ -0,0 +1,73 @@
+
+A sample script will look like:
+```
+import ngraph_bridge
+
+in0, in1, out0 = construct_network()
+sess = tf.Session(config=ngraph_bridge.update_config(tf.ConfigProto()))
+result = sess.run(out0, feed_dict = {in0:[2,3], in1:[4,5]})
+```
+
+Notice the new line introduced here wrt the "normal" path. `config_updated = ngraph_bridge.update_config(config)` must now be passed during session construction to ensure `NgraphOptimizer` `grappler` pass is enabled. Like the "normal" path, in this script too the graph rewriting and actual execution happen when `session.run` is called.
+
+
+### tf2ngraph
+
+The advantage of `grappler` is that the graph rewriting can happen before `session.run` is called. This is exactly what `tf2ngraph` does. The workflow can now look like this:
+```
+python tf2ngraph.py --inputpbtxt input.pbtxt --outnodes out_node --outputpbtxt output.pbtxt --ngbackend CPU
+```
+followed by
+```
+import ngraph_bridge
+from google.protobuf import text_format
+
+with open('output.pbtxt', "r") as f:
+    text_format.Merge(f.read(), graph_def)
+sess = tf.Session(graph=graph)
+in0 = get_tensor_by_name('in0')
+in1 = get_tensor_by_name('in1')
+out0 = get_tensor_by_name('out0')
+result = sess.run(out1, feed_dict={in0:[2,3], in1:[4,5]})
+```
+
+Note that in the second phase, we did not need `ngraph_bridge.update_config`. This is because all graph rewrites have finished in the first phase (`tf2ngraph`) and now in the second phase the `import ngraph_bridge` is needed only to identify the custom op `NGraphEncapsulate`, but not for rewrites. If we set the environment flag `NGRAPH_TF_LOG_PLACEMENT=1` in the first phase, we would see the TF nodes with some being placed on `ngraph`. If we use `NGRAPH_TF_LOG_PLACEMENT=1` in the second phase, we would not see most `tensorflow` nodes since they have disappeared inside `NGraphEncapsulate`s. We will only see a few `tensorflow` nodes (those unsupported by `ngraph-bridge` hence not captured inside `NGraphEncapsulate`) and some `NGraphEncapsulate` nodes. All nodes will be shown placed on `HOST` and not on `ngraph` (because `NGraphEncapsulate` runs on `HOST`)
+
+Some features of `tf2ngraph:
+1. You can get a comprehensive help string using: `python tf2ngraph.py -h`
+2. It supports 3 input formats: `pb`, `pbtxt` and `saved model`
+3. It supports 3 output formats: `pb`, `pbtxt` and `saved model`
+4. It requires the user to specify the output node names using `--outnodes`. For example: `--outnodes add:0,sub:0`
+5. Optionally one can also specify the backend along with config strings after colons `--ngbackend NNPI` or `--ngbackend GPU:0` etc. By default we use CPU
```

Comment:
Should we not specify NNPI here since we would want them to use the rewrite config to specify the backend or may be mention about specifying the backend along with different config parameters using the rewrite config?

Reply:
change to --ngbackend CPU, let it be a generic example